db/migrate/20230719080900_fix_ubuntu_versions.rb

```diff
@@ -0,0 +1,12 @@
+class FixUbuntuVersions < ActiveRecord::Migration[6.0]
+  def up
+    # For every Ubuntu operating system with a set minor version we
+    #   * Check if there is already an operating system present with the same major and minor version combined in the major field
+    #   * If not, we update the version
+    Operatingsystem.where(name: 'Ubuntu').where.not(minor: [nil, '']).each do |os|
+      unless Operatingsystem.find_by(name: 'Ubuntu', major: "#{os.major}.#{os.minor}", minor: [nil, ''])
```

Comment:
In general it's a bad practice to run an SQL query for every item, which results in 1+n queries. Here it may not be too bad because it's likely a small result set and a migration only runs once.

Reply:
I converted the db migration to a single joined update query and tested the behavior with following content in the `operatingsystems` table:

```
 id | major |    name     | minor |        id | major |    name     | minor |
----+-------+-------------+-------+       ----+-------+-------------+-------+
  1 | 18    | Ubuntu      | 04    |         1 | 18.04 | Ubuntu      |       |
  2 | 20.04 | Ubuntu      |       |   =>    2 | 20.04 | Ubuntu      |       |
  3 | 22.04 | Ubuntu      | 3     |         3 | 22.04 | Ubuntu      | 3     |
  4 | 20    | Ubuntu      | 04    |         4 | 20    | Ubuntu      | 04    |
```